sci-mathematics/mathematica/mathematica-13.0.1.ebuild

```diff
@@ -0,0 +1,155 @@
+		-o -name MacOSX -o -name MacOSX-x86-64 -o -name Macintosh -\) \
+		-exec rm -rv {} + || die
+
+	if ! use cuda; then
+		einfo 'Removing cuda support'
+		rm -r "${S}/${M_TARGET}/SystemFiles/Components/CUDACompileTools/LibraryResources/Linux-x86-64/CUDAExtensions.so" || die
+	fi
+
+	# Linux-x86-64/AllVersions is the supported version, other versions remove
+	einfo 'Removing unsupported RLink versions'
+	rm -r "${S}/${M_TARGET}/SystemFiles/Links/RLink/SystemFiles/Libraries/Linux-x86-64/3.5.0" || die
+	rm -r "${S}/${M_TARGET}/SystemFiles/Links/RLink/SystemFiles/Libraries/Linux-x86-64/3.6.0" || die
+	rm -r "${S}/${M_TARGET}/SystemFiles/Links/RLink/SystemFiles/Libraries/Linux/AllVersions" || die
+	if ! use R; then
+		einfo 'Removing RLink support'
+		rm -r "${S}/${M_TARGET}/SystemFiles/Links/RLink/SystemFiles/Libraries/Linux-x86-64/AllVersions/libjri.so" || die
+	fi
+
+	# fix RPATH
+	while IFS= read -r -d '' i; do
+		# Use \x7fELF header to separate ELF executables and libraries
+		# Skip .o files and static files to avoid surprises
+		[[ $(od -t x1 -N 4 "${i}") == *"7f 45 4c 46"* ]] || continue
+		[[ -f "${i}" && "${i: -2}" != ".o" ]] || continue
+		[[ "$(file "${i}")" == *"dynamically"* ]] || continue
+		einfo "Fixing RPATH of ${i}"
+		patchelf --set-rpath \
+'/'"${M_TARGET}"'/SystemFiles/Libraries/Linux-x86-64:'\
+'/'"${M_TARGET}"'/SystemFiles/Libraries/Linux-x86-64/Qt/lib:'\
+'/'"${M_TARGET}"'/SystemFiles/Java/Linux-x86-64/lib:'\
+'/'"${M_TARGET}"'/SystemFiles/Java/Linux-x86-64/lib/jli:'\
+'$ORIGIN' "${i}" || \
+		die "patchelf failed on ${i}"
+	done < <(find "${S}/${M_TARGET}" -type f -print0)
+
+	# fix broken symbolic link
+	ln -sf "/${M_TARGET}/SystemFiles/Kernel/Binaries/Linux-x86-64/wolframscript" "${S}/${M_TARGET}/Executables/wolframscript" || die
+
+	# move all over
+	mv "${S}"/opt "${ED}"/opt || die
+
+	# the autogenerated symlinks point into sandbox, remove
+	rm "${ED}"/opt/bin/* || die
+
+	# install wrappers instead
+	for name in ${M_BINARIES} ; do
+		einfo "Generating wrapper for ${name}"
+		echo '#!/bin/sh' >> "${T}/${name}" || die
+		echo 'QT_QPA_PLATFORM="wayland;xcb"' >> "${T}/${name}" || die
+		echo "LD_PRELOAD=/usr/$(get_libdir)/libfreetype.so.6:/$(get_libdir)/libz.so.1:/$(get_libdir)/libcrypt.so.1 /${M_TARGET}/Executables/${name} \$*" \
+			>> "${T}/${name}" || die
+		dobin "${T}/${name}"
+	done
+	for name in ${M_BINARIES} ; do
+		einfo "Symlinking ${name} to /opt/bin"
+		dosym ../../usr/bin/${name} /opt/bin/${name}
+	done
+
+	# fix some embedded paths and install desktop files
+	for filename in $(find "${ED}/${M_TARGET}/SystemFiles/Installation" -name "wolfram-mathematica*.desktop") ; do
+		einfo "Fixing ${filename}"
+		sed -e "s|${S}||g" -e 's|^\t\t||g' -i "${filename}" || die
+		echo "Categories=Physics;Science;Engineering;2DGraphics;Graphics;" >> "${filename}" || die
+		domenu "${filename}"
+	done
+
+	# install icons
+	doicon -s scalable "${FILESDIR}/wolfram-mathematica.svg"
```

Comment:
I'm still concerned that we can't legally have the svg icon file in the repository.

Instead of using `doicon` on an icon in FILESDIR, could we use `${ED}/opt/Wolfram/Mathematica/13.0/SystemFiles/FrontEnd/SystemResources/X/App-{32,64,128}.png` instead? That way we don't have to have the icon file in the repository.

Reply:
Done